src/pages/installation.md

```diff
@@ -0,0 +1,181 @@
+
+You can install Astro via the **CLI**. This way you can kick off with some starter projects or view some demo projects to get better acquainted with Astro.
+
+First create your directory that we would be deploying Astro in.
+
+```bash
+mkdir my-astro-project
+```
+
+Inside your directory, we would first install Astro using your preferred package manager:
+
+```bash
+npm init astro || yarn add astro
+```
+
+You would be asked if you wish to download the Astro packages to the System, enter `Y` to accept.
+
+Next our **CLI tool** would start up to walk you through the rest of installation.
+
+It would ask you which *template* you wish to use, here we have a few options for you to choose from to get started.
+
+- **Starter Kit** - This is our base project that allows you to take Astro and start building anything you wish
+- **Blog** - Astro's Blog template. This will let you explore and build your own Astro blogging site
+- **Documentation** - Astro's Doc's template, see how Astro builds and creates Documentation Sites
+- **Portfolio** - Astro's Portfolio Starter Template. Here  you can experiment and build professional portfolio sites with Astro
+<!-- I think we should have guides for each of these templates, perhaps in a template section, with tutorials on building something with each of them, -->
+
+We would recommend to start with the 'Starter Kit' template. This would give us a solid platform to liftoff from.
+
+Astro will begin to create and setup the project files for you, once done you will be prompted with the post-installation next steps.
+
+```bash
+  1: npm install (or pnpm install, yarn, etc)
+  2: git init && git add -A && git commit -m "Initial commit" (optional step)
+  3: npm start (or pnpm, yarn, etc)
+```
+
+At this stage, Astro has merely been copied to your new project. We would need to install the Astro dependencies and setup our `node_modules` folder inside the project.
+
+```bash
+npm install || yarn
+```
+
+This process will first, fetch the required dependencies from the npm registry, and then link and build the packages out for Astro to be able to use.
+
+```bash
+git init && git add -A && git commit -m "Initial commit"
+```
+
+This step, is optional, we do recommend setting up and adding `git` to the project. This combined command will let you set this up with ease.
+
+Astro adds only two commands to the `scripts` in the `package.json` file:
+
+```js
+{
+  "scripts": {
+    "start": "astro dev",
+    "build": "astro build"
+  }
+}
+
+```
+To develop your site in Astro, run the following command:
+
+```bash
+npm run start || yarn start || pnpm start
+```
+
+It starts the Astro **dev** server. This would start [Snowpack](https://snowpack.dev), which prepares all the Astro dependencies. This is a 'one-time' step and the results of which would be cached for the lifetime of the project.
+
+Once Snowpack has completed its task, you would be able to view the `dev` server at the default address of: `http://127.0.0.1:3000/`
```

Comment:
> default address of: `http://127.0.0.1:3000/`
maybe we can **also** add `http://localhost:3000` here.

Reply:
I would like this also, however, in the console, it appears as the `127...` , I figured that might throw some people, If this could be altered on the CLI then we could reflect the docs to the 'localhost' .... also of topic, what about docker? - this I feel would be put into the deployment category, however, a small docker install here would also be beneficial.

Reply:
Huuuum, I think for the deployment it makes sense, but here, not really, since docker is more when you have the application "ready" if it's just for people to see, sure, but if for development makes not much sense here.

Reply:
totally, awesome, the deployment section is going to fun to write